websites/W/WebProfiles/presence.ts

```diff
@@ -4,25 +4,39 @@ const presence = new Presence({
 	browsingTimestamp = Math.floor(Date.now() / 1000);
 
 presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {
+		largeImageKey: "webprofiles_logo",
+	};
+
 	const showTimestamp = await presence.getSetting<boolean>("timestamp"),
 		showButtons = await presence.getSetting<boolean>("buttons"),
-		presenceData: PresenceData = {
-			largeImageKey: "webprofiles_logo",
+		pages: Record<string, PresenceData> = {
+			"/about/team": { details: "Viewing page:", state: "Team" },
+			"/about/partners": { details: "Viewing page:", state: "Partners" },
+			"/discover": { details: "Viewing page:", state: "Discover" },
+			"/@me/settings": { details: "Editing my profile" },
+			"/login": { details: "Logging in..." },
+			register: { details: "Registering..." },
 		};
 
+	for (const [path, data] of Object.entries(pages)) {
+		if (document.location.pathname.includes(path))
+			presenceData = { ...presenceData, ...data };
+	}
+
 	if (document.location.pathname === "/")
 		presenceData.details = "Viewing home page";
 	else if (document.location.pathname.includes("/u/")) {
 		const username = document.querySelector(
 			"p.text-5xl.text-white"
 		)?.textContent;
-		presenceData.details = "Viewing user:";
-		presenceData.state = `${username || "Unknown"} - ❤️ ${
-			document.querySelector("#likes-count")?.textContent || 0
-		}`;
+		presenceData.details = `Viewing user: ${username || "Unknown"}`;
+		presenceData.state = `${
+			document.querySelector("p.mt-3")?.textContent
+		} - ❤️ ${document.querySelector("#likes-count")?.textContent || 0}`;
 		presenceData.buttons = [
 			{
-				label: `View ${username}`,
+				label: `View ${username} Profile`,
```

Comment:
Ideally buttons should **not** be dynamic, because:
- Repetitive of details above
- Could exceed Discord limits

```suggestion
				label: `View Profile`,
```